tests/src/controller/checkout.controller.test.ts

```diff
@@ -116,4 +123,96 @@ describe('Checkout controller', () => {
       });
     });
   });
+
+  describe('Creating a subscription', () => {
+    it('When the user wants to create a subscription, it is created successfully', async () => {
+      const mockedUser = getUser();
+      const mockedSubscription = getCreatedSubscription();
+      const mockedSubscriptionResponse = getCreateSubscriptionResponse();
+
+      const authToken = getValidAuthToken(mockedUser.uuid);
+      const userToken = getValidUserToken(mockedUser.customerId);
+
+      jest.spyOn(PaymentService.prototype, 'createSubscription').mockResolvedValue(mockedSubscriptionResponse);
+
+      const response = await app.inject({
+        path: '/checkout/subscription',
+        method: 'POST',
+        body: {
+          customerId: mockedUser.customerId,
+          priceId: mockedSubscription.items.data[0].price.id,
+          currency: mockedSubscription.items.data[0].price.currency,
+          quantity: 1,
+          token: userToken,
+        },
+        headers: {
+          authorization: `Bearer ${authToken}`,
+        },
+      });
+
+      const responseBody = response.json();
+
+      expect(response.statusCode).toBe(200);
+      expect(responseBody).toStrictEqual(mockedSubscriptionResponse);
+    });
+
+    describe('Handling errors', () => {
+      it('When the id of the price is not present in the body, then an error indicating so is thrown', async () => {
+        const mockedUser = getUser();
+        const authToken = getValidAuthToken(mockedUser.uuid);
+
+        const response = await app.inject({
+          path: '/checkout/subscription',
+          method: 'POST',
+          body: {
+            customerId: mockedUser.customerId,
+          },
+          headers: {
+            authorization: `Bearer ${authToken}`,
+          },
+        });
+
+        expect(response.statusCode).toBe(400);
+      });
+
+      it('When the id of the customer is not present in the body, then an error indicating so is thrown', async () => {
+        const mockedUser = getUser();
+        const authToken = getValidAuthToken(mockedUser.uuid);
+
+        const response = await app.inject({
+          path: '/checkout/subscription',
+          method: 'POST',
+          body: {
+            priceId: 'price_id',
+          },
+          headers: {
+            authorization: `Bearer ${authToken}`,
+          },
+        });
+
+        expect(response.statusCode).toBe(400);
+      });
+
+      it('When the provided user token is not valid, then an error indicating so is thrown', async () => {
```

Comment:
This test made my think about the diff between not a valid token and a non-matching customer Id (`if (customerId !== tokenCustomerId)` ....). You are testing the second, what about the first? Also, you may reconsider the legend of this test as the token is valid in this case, just contains a different customer id (either an impersonation attempt or a code logic error in the frontend)